test-data/unit/check-isinstance.test

```diff
@@ -2065,6 +2065,18 @@ def f() -> None:
 [builtins fixtures/dict.pyi]
 [out]
 
+[case testNarrowTypeAfterInDictKeys]
```

Comment:
I just added this here because there was a similar test just above for `TypedDict`s